src/actions/grant.js

```diff
@@ -127,16 +138,19 @@ export function submitGrant() {
         dispatch(showValidationErrors({ form: errorMessages }));
         return false;
       }
+
+      const submissionId = data.createSubmission.id;
+      log('success!');
+      dispatch(submitGrantSuccess(submissionId));
+      history.push('/grant/success');
```

Comment:
I will make some changes that might shine some light on solutions for what you are thinking